emission/analysis/userinput/matcher.py

```diff
@@ -27,29 +28,52 @@ def match_incoming_user_inputs(user_id):
 
 def match_incoming_inputs(user_id, timerange):
     ts = esta.TimeSeries.get_time_series(user_id)
-    input_key_list = eac.get_config()["userinput.keylist"]
+    single_most_recent_match_key_list = eac.get_config()["userinput.keylist"]
+    multi_non_deleted_key_list = ["manual/trip_addition_input"]
+    input_key_list = single_most_recent_match_key_list + multi_non_deleted_key_list
     toMatchInputs = [ecwe.Entry(e) for e in ts.find_entries(input_key_list, time_query=timerange)]
-    logging.debug("Matching %d inputs to trips" % len(toMatchInputs))
+    logging.debug("Matching %d single inputs to trips" % len(toMatchInputs))
     lastInputProcessed = None
     if len(toMatchInputs) == 0:
         logging.debug("len(toMatchInputs) == 0, early return")
         return None
     for ui in toMatchInputs:
```

Comment:
I have been having a hard time understanding how input matching works on phone vs server. I now understand why:

I see here that input matching on the server works by iterating through each user input (ui), and then ui is matched to the appropriate trip by esdt.get_trip_for_user_input_obj

This is fundamentally different on the phone: input matching works by iterating through each trip, and then finds the user inputs that correspond to that trip (with the functions getUserInputForTrip and getTripAdditionsForTrip in input-matcher.js)

If we want to unify the matching functions, and improve interoperability and maintainability, we would have to reconcile that fundamental difference

Reply:
@JGreenlee the server actually does both:
- in the common case, in which the user labels processed trips, we do indeed match the inputs to the trips
- however, for the uncommon case, in which the user labels draft trips, and we process both the trip information and the inputs at the same time, we match the trip to the inputs. Please see https://github.com/e-mission/e-mission-server/pull/895/commits/a7ee02c6e689f00368f88a219fd8d21c2da896ec